src/lune/builtins/process/mod.rs

```diff
@@ -202,14 +205,32 @@ async fn spawn_command(
     options: ProcessSpawnOptions,
 ) -> LuaResult<(ExitStatus, Vec<u8>, Vec<u8>)> {
     let inherit_stdio = options.inherit_stdio;
+    let stdin = options.stdin.clone();
 
-    let child = options
+    let mut child = options
         .into_command(program, args)
-        .stdin(Stdio::null())
+        .stdin(match stdin.is_some() {
+            true => Stdio::piped(),
+            false => Stdio::null(),
+        })
         .stdout(Stdio::piped())
         .stderr(Stdio::piped())
         .spawn()?;
 
+    // If the stdin option was provided, we write that to the child
+    if let Some(stdin) = stdin {
+        let mut child_stdin = child.stdin.take().unwrap();
+
+        let _ = task::spawn(async move {
+            let mut tee = AsyncTeeWriter::new(&mut child_stdin);
+            tee.write_all(stdin.as_bytes()).await.into_lua_err()?;
+
+            Ok::<(), LuaError>(())
+        })
+        .await
+        .expect("Tee writer for stdin errored");
```

Comment:
```suggestion
        let mut child_stdin = child.stdin.take().unwrap();
        child_stdin
            .write_all(stdin.as_bytes())
            .await
            .into_lua_err()?;
```

Two things here:

- `AsyncTeeWriter` is not needed, it is used when we need to duplicate writing, when we need to both store the result of `stdout` in a string while also outputting it to stdout as we go (the "inherit" option)
- Spawning a new task is not necessary it is `await`-ed it right away

Reply:
Aha, my bad. I'm not really experienced in async rust, so apologies for that.